lib/internal/modules/esm/resolve.js

```diff
@@ -249,7 +249,7 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
 
   const stats = internalFsBinding.internalModuleStat(
     internalFsBinding,
-    StringPrototypeEndsWith(internalFsBinding, path, '/') ? StringPrototypeSlice(path, -1) : path,
+    StringPrototypeEndsWith(path, '/') ? StringPrototypeSlice(path, -1) : path,
```

Comment:
this definitely seems like a correct change, because string ends with only takes two arguments, the receiver and the substring - but i still think it needs a test (that would have failed without the fix)